Doc/library/atexit.rst

```diff
@@ -14,11 +14,11 @@ functions.  Functions thus registered are automatically executed upon normal
 interpreter termination.  :mod:`atexit` runs these functions in the *reverse*
 order in which they were registered; if you register ``A``, ``B``, and ``C``,
 at interpreter termination time they will be run in the order ``C``, ``B``,
-``A``.
-
-**Note:** The functions registered via this module are not called when the
+``A``.  The functions registered via this module are not called when the
 program is killed by a signal not handled by Python, when a Python fatal
-internal error is detected, or when :func:`os._exit` is called.
+internal error is detected, or when :func:`os._exit` is called.  Equality
+comparisons are used internally during unregistration operations, so function
+references do not need to have matching identities.
```

Comment:
I think it might fit better in the unregister section, which may need to be reworded a bit. (For instance, I don't like the guarantee that func will not be called at shutdown - what if it's registered again?). How about something like:

Remove from the list of functions to be run at interpreter shutdown all registrations of functions that compare equal (``==``) to *func*. If no such registered functions exist, :func:`unregister` silently does nothing.

Reply:
I like that you removed the **Note** here though.

Reply:
Good point about moving that into the unregister section! I agree that the "guarantee" statement is confusing and unnecessary. However, I prefer to keep three separate sentences for better clarity. What do you think?

Reply:
Also, what do you think about inserting one more sentence into the unregister section:

> If *func* has been registered more than once, every occurrence of that function in the :mod:`atexit` call stack will be removed.

Reply:
Yes, that needs to be mentioned, this is why I worded it "remove all fs that compare equal to func" rather than just "remove func, multiple times".

Reply:
^^ Information mentioned in cd99401